conda/cli/common.py

```diff
@@ -263,3 +266,37 @@ def validate_prefix(prefix):
         raise EnvironmentLocationNotFound(prefix)
 
     return prefix
+
+
+def validate_prefix_is_writable(prefix) -> str:
+    """Verifies the environment directory is writable.
+
+    :raises EnvironmentNotWritableError: Conda does not have permission to write to the prefix
+    :returns: Valid prefix.
+    :rtype: str
+    """
+    # ensure conda access to the prefix dir
+    if not os.access(prefix, os.W_OK):
```

Comment:
[Other places in the code](https://github.com/conda/conda/blob/main/conda/core/link.py#L774-L790) conda just checks if the history file is writable. So, this is maybe not needed?